### PR TITLE
Blog app views

### PR DESCRIPTION
**Pull request by AhmedNazirMusah**
On this milestone, I implemented the following requirement.
- Implement the design from the sneak peek wireframes.
Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in the coming milestone)
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- Hint: In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+      <h3><%= post.title %></h3>
+      <p><%= post.text %></p>
+      <span>
+        Comments: <%= post.comments_counter %>
+        Likes: <%= post.likes_counter %>
+      </span>
+    </div>
+  <% end %>
+</div>
+<a class="btn" href="/users/<%= @single_user.id %>/posts">See all posts</a>
```
Comment (tresorsawasawa) [inline app/views/users/show.html.erb]: - [ ] Kindly make sure when clicking the `See all Post` link it opens `user's post index` as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png). Right now, this is the error I am getting:
Screenshot from 2023-01-27 04-24-15

File: app/views/Posts/index.html.erb [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+        </span>
+      </div>
+      <div class="post-comment">
+        <% post.recent_comments.each do |comment| %>
+          <h3><%= comment.author.name %>: <%= comment.text%></h3>
+        <% end %>
+      </div>
+    </div>
+  <% end %>
+</div>
```
Comment (tresorsawasawa) [inline app/views/Posts/index.html.erb]: - [ ] Kindly ensure that the `app/views/Posts/index.html.erb` view page is working as it should display this **[wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png)**. Right now it shows the below error, so impossible to check the result:
Screenshot from 2023-01-27 04-24-15

File: app/views/Posts/show.html.erb [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
-  <body class="body">
-    <h1>Show method for posts</h1>
-  </body>
-</html>
\ No newline at end of file
+<div class="post-comment">
+  <% @post_comments.each do |comment| %>
+    <span><%= comment.author.name %>: <%= comment.text %></span> 
+  <% end %>
+</div>
```
Comment (tresorsawasawa) [inline app/views/Posts/show.html.erb]: - [ ] As I mentioned in the previous comment, kindly ensure that the `app/views/Posts/show.html.erb` view page is working as it should display this **[wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png)**(i.e. Post show page should display post with all comments listed as shown on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_single_post.png)). Right now it shows the below error, so impossible to check the result:
Screenshot from 2023-01-27 04-28-21

**In short, any page for "Post" is working, all pages show the similar error. Make sure to fix that**

File: app/views/Posts/index.html.erb [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+          Comments: <%= post.comments_counter %>
+          Likes: <%= post.likes_counter %>
+        </span>
+      </div>
+      <div class="post-comment">
+        <% post.recent_comments.each do |comment| %>
+          <h3><%= comment.author.name %>: <%= comment.text%></h3>
+        <% end %>
+      </div>
+    </div>
```
Comment (tresorsawasawa) [inline app/views/Posts/index.html.erb]: - [ OPTIONAL ] I think it would be better if you use partial files on a specific Post for best practices. I suggest you refer this [article](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) to get a short reminder and examples

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -1,16 +1,11 @@
...
+  <a class="user-link" href="/users/<%= user.id%>">
+  <section class="all-users">  
+    <div class="img-holder"><%= user.photo %></div>
+      <div class="user-info">
+        <h2><%= user.name %></h2> 
+        <span>Number Of Posts: <%= user.post_counter %></span>
+      </div>
+    </section>
+  </a>
+<% end %>
```
Comment (tresorsawasawa) [inline app/views/users/index.html.erb:11]: - [ OPTIONAL ] I think it would be better if you use partial files on a specific Post for best practices. I suggest you refer this [article](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) to get a short reminder and examples

File: app/views/Posts/index.html.erb (hunk 4 above)
Comment (tresorsawasawa) [inline app/views/Posts/index.html.erb]: - [ **OPTIONAL** ] I think it would be better to add a feature when clicking on each `Post` takes you to its show page as shown in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png) as you did for a `User`

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @AhmedNazirMusah 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/users/show.html.erb]: Needs attention!

File: app/views/Posts/index.html.erb (hunk 2 above)
Comment (rloterh) [reply app/views/Posts/index.html.erb]: Needs attention!

File: app/views/Posts/show.html.erb (hunk 3 above)
Comment (rloterh) [reply app/views/Posts/show.html.erb]: Needs attention!

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -1,5 +1,11 @@
...
+  def index
+    @single_user = User.find(params[:user_id].to_i)
+    @user_posts = Post.where(author: @single_user)
+  end
 
-  def show; end
+  def show
+    @single_post = Post.find(params[:id].to_i)
+    @post_comments = Comment.where(post: @single_post)
+  end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:10]: - [ ] Attempt to show single or all posts by a user throws error, like mentioned in the previous review. Kindly ensure to the show action is directed to the appropriate view template.

**View post error:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ahmed, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦   🏃

## Highlights 👍  
- Good job with views implementations so far!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nicupop729 — COMMENTED** [review]
Hello @AhmedNazirMusah  👋,

_This is an additional review as a result of the second opinion request._

I appreciate a lot your great effort for this milestone! I know, is not an easy one.
You are almost there!

Unfortunately, I must confirm what the previous reviewers mentioned.

Let's have a look on what at the moment your app displays:
- for http://127.0.0.1:3000 it displays correctly the users (we have Lill and Lilly, cause I was lazy to come with different names, and mostly I used the data from your README to populate the database)
image
- for http://127.0.0.1:3000/users/1 we have Lilly, correctly displayed
image
- for http://127.0.0.1:3000/users/1/posts I should have all the posts of Lilly, but unfortunatelly, I have this
image
- for http://127.0.0.1:3000/users/1/posts/1 I should the first post of Lilly, but yet, unfortunatelly, I have this
image

Kindly make sure to respect the wireframe provided by Microverse and also, make sure to display at each route the corresponding info requested.
It will be useful to give correct functionality to the See all posts button too, and to follow all the optional suggestions asked by the first reviewer.

Please take some time understand the flow of information in your app and try your best to implement them. Remember: your peers may be a good source of help for this project.

When ready, kindly ask for another review in the dashboard.

Keep rocking 🚀!

File: app/views/posts/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<%= render "shared/user"%>
```
Comment (MeqdamAlqudah) [inline app/views/posts/index.html.erb:1]: - [Optional] Good Job I can see all the links as was requested by the last reviewer 
image
image
image
image
but I noticed that the post count isn't updated unless the user makes a comment on the post !! Please fix this 
image

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @AhmedNazirMusah , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## HighLight 
:+1: no linter errors 
:+1: links are working 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
